src/pybind/mgr/dashboard/controllers/iscsi.py

```diff
@@ -938,3 +954,23 @@ def validate_rest_api(gateways):
                                          '{}'.format(gateway),
                                      code='ceph_iscsi_rest_api_not_available_for_gateway',
                                      component='iscsi')
+
+
+def validate_auth(auth):
+    username_regex = re.compile(r'^[\w\.:@_-]{8,64}$')
+    password_regex = re.compile(r'^[\w@\-_\/]{12,16}$')
+    result = True
+
+    if auth['user'] or auth['password']:
+        result = result and bool(username_regex.match(auth['user']))
+        result = result and bool(password_regex.match(auth['password']))
```

Comment:
There is no need to assign it twice or more in every if-clause.

```suggestion
        result = result and bool(username_regex.match(auth['user'])) and bool(password_regex.match(auth['password']))
```